core/src/main/java/io/grpc/internal/RetriableStream.java

```diff
@@ -64,6 +64,7 @@
 
   private final MethodDescriptor<ReqT, ?> method;
   private final Executor callExecutor;
+  private final Executor listenerSerializeExecutor = new SerializeReentrantCallsDirectExecutor();
```

Comment:
SerializeReentrantCallsDirectExecutor is single-producer (not thread safe). It doesn't work in this case. I think SynchronizationContext is probably better suited.

Reply:
Done.